code/modules/mob/living/carbon/carbon_life.dm

```diff
@@ -285,6 +286,27 @@
 		else
 			healths.icon_state = "health7"
 
+/mob/living/carbon/update_stamina_hud()
+	if(!client)
+		return
+	if(staminas)
+		var/perceived_stamina = staminaloss - shock_reduction()
```

Comment:
It feels to me like you could probably add a proc to determine the perceived stamina on the `mob/living` level (and override it here on the carbon level), and then just remove this override in favor of a single implementation

Reply:
Apologies fort he long wait, done!